File: api/javer/src/main/java/com/uuhnaut69/javer/resource/RestaurantResource.java
package com.uuhnaut69.javer.resource;

import com.uuhnaut69.javer.model.Dish;
import com.uuhnaut69.javer.model.Restaurant;
import com.uuhnaut69.javer.resource.dto.DishRequest;
import com.uuhnaut69.javer.resource.dto.RestaurantRequest;
import com.uuhnaut69.javer.service.DishService;
import com.uuhnaut69.javer.service.RestaurantService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;
import java.util.List;
import java.util.UUID;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/restaurants")
public class RestaurantResource {

  private final DishService dishService;

  private final RestaurantService restaurantService;

  @PostMapping
  @ResponseStatus(HttpStatus.CREATED)
  public Restaurant createRestaurant(@RequestBody @Valid RestaurantRequest restaurantRequest) {
    return restaurantService.create(restaurantRequest);
  }

  @GetMapping
  public List<Restaurant> findAllRestaurants() {
    return restaurantService.findAll();
  }

  @PutMapping("/{restaurantId}")
  public Restaurant updateRestaurant(
      @PathVariable UUID restaurantId, @RequestBody @Valid RestaurantRequest restaurantRequest) {
    return restaurantService.update(restaurantId, restaurantRequest);
  }

  @DeleteMapping("/{restaurantId}")
  @ResponseStatus(HttpStatus.NO_CONTENT)
  public void deleteRestaurant(@PathVariable UUID restaurantId) {
    restaurantService.delete(restaurantId);
  }

  @PostMapping("/{restaurantId}/dishes")
  @ResponseStatus(HttpStatus.CREATED)
  public Dish createDish(
      @PathVariable UUID restaurantId, @RequestBody @Valid DishRequest dishRequest) {
    return dishService.create(restaurantId, dishRequest);
  }

  @PutMapping("/{restaurantId}/dishes/{dishId}")
  public Dish updateDish(
      @PathVariable UUID restaurantId,
      @PathVariable UUID dishId,
      @RequestBody @Valid DishRequest dishRequest) {
    return dishService.update(restaurantId, dishId, dishRequest);
  }

  @DeleteMapping("/{restaurantId}/dishes/{dishId}")
  @ResponseStatus(HttpStatus.NO_CONTENT)
  public void deleteDish(@PathVariable UUID restaurantId, @PathVariable UUID dishId) {
    dishService.delete(restaurantId, dishId);
  }

  @GetMapping("/{restaurantId}/dishes")
  public List<Dish> findAllByRestaurantId(@PathVariable UUID restaurantId) {
    return dishService.findAllByRestaurantId(restaurantId);
  }
}

File: api/kover/src/main/kotlin/com/uuhnaut69/kover/resource/RestaurantResource.kt
package com.uuhnaut69.kover.resource

import com.uuhnaut69.kover.model.Dish
import com.uuhnaut69.kover.model.Restaurant
import com.uuhnaut69.kover.resource.dto.DishRequest
import com.uuhnaut69.kover.resource.dto.RestaurantRequest
import com.uuhnaut69.kover.service.DishService
import com.uuhnaut69.kover.service.RestaurantService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import java.util.*
import javax.validation.Valid

@RestController
@RequestMapping("/api/restaurants")
class RestaurantResource(private val dishService: DishService, private val restaurantService: RestaurantService) {

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun createRestaurant(@RequestBody @Valid restaurantRequest: RestaurantRequest): Restaurant =
        restaurantService.create(restaurantRequest)

    @GetMapping
    fun findAllRestaurants(): List<Restaurant> = restaurantService.findAll()

    @PutMapping("/{restaurantId}")
    fun updateRestaurant(
        @PathVariable restaurantId: UUID,
        @RequestBody @Valid restaurantRequest: RestaurantRequest
    ): Restaurant = restaurantService.update(restaurantId, restaurantRequest)

    @DeleteMapping("/{restaurantId}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteRestaurant(
        @PathVariable restaurantId: UUID
    ) = restaurantService.delete(restaurantId)

    @PostMapping("/{restaurantId}/dishes")
    @ResponseStatus(HttpStatus.CREATED)
    fun createDish(@PathVariable restaurantId: UUID, @RequestBody @Valid dishRequest: DishRequest): Dish =
        dishService.create(restaurantId, dishRequest)

    @PutMapping("/{restaurantId}/dishes/{dishId}")
    fun updateDish(
        @PathVariable restaurantId: UUID,
        @PathVariable dishId: UUID,
        @RequestBody @Valid dishRequest: DishRequest
    ): Dish = dishService.update(restaurantId, dishId, dishRequest)

    @DeleteMapping("/{restaurantId}/dishes/{dishId}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteDish(
        @PathVariable restaurantId: UUID,
        @PathVariable dishId: UUID
    ) = dishService.delete(restaurantId, dishId)

    @GetMapping("/{restaurantId}/dishes")
    fun findAllByRestaurantId(@PathVariable restaurantId: UUID): List<Dish> =
        dishService.findAllByRestaurantId(restaurantId)
}